src/pages/Register/index.js

```diff
@@ -1,27 +1,93 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
 
 import LoginLogo from "../../assets/LoginLogo.png";
 import ProjectLogo from "../../assets/ProjectLogo.png";
+import { validateCPF, validateEmail } from '../../utils/validators';
+import { registerCitizen } from "../../service/citizen.service";
+import { registerLegislator } from "../../service/legislator.service";
 
 import './styles.css';
 
 export default function Register() {
-    const [nextStep, setNextStep] = useState(4);
     const [firstStep, setFirstStep] = useState(true);
+    const [name, setName] = useState("")
+    const [cpf, setCpf] = useState("")
+    const [state, setState] = useState("")
+    const [city, setCity] = useState("")
+    const [email, setEmail] = useState("")
+    const [password, setPassword] = useState("")
+    const [passwordConfirmation, setPasswordConfirmation] = useState("")
+    const [filiation, setFiliation] = useState("")
     const [isCitizen, setIsCitizen] = useState(true);
+    const [isFormCompleted, setIsFormCompleted] = useState(false);
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if (isCitizen) {
+            if (!!name && cpf.length === 11 && !!state && !!city && !!email && !!password && !!passwordConfirmation) {
+                let isValid = true;
+                isValid = validateEmail(email) && validateCPF(cpf) && (password === passwordConfirmation);
+
+                setIsFormCompleted(isValid);
+            } else {
+                setIsFormCompleted(false);
+            }
```

Comment:
Já que existe a função de validateCpf, a checagem de length pode ficar por conta dessa função.
```suggestion
            if (!!name && !!state && !!city && !!email && !!password && !!passwordConfirmation) {
                let isValid = true;
                isValid = validateEmail(email) && validateCPF(cpf) && (password === passwordConfirmation);

                setIsFormCompleted(isValid);
            } else {
                setIsFormCompleted(false);
            }
```

Reply:
Concordo plenamente com esse ponto. Vou ajustar hoje!